daft/runners/ray_runner.py

```diff
@@ -738,9 +739,16 @@ def __init__(
         max_task_backlog: int | None,
     ) -> None:
         super().__init__()
+
+        # Reuse the existing Ray context if already initialized
         if ray.is_initialized():
-            logger.warning("Ray has already been initialized, Daft will reuse the existing Ray context.")
-        self.ray_context = ray.init(address=address, ignore_reinit_error=True)
+            if address is not None:
+                warnings.warn(
+                    f"Ray has already been initialized. Daft will default to using the existing Ray connection and ignore the supplied address: {address}"
+                )
+            self.ray_context = ray.init(ignore_reinit_error=True)
```

Comment:
One last TODO: figure out how to not call `ray.init` here again, which is weird when a user passes in `None` e.g.

```
ray.init(address="ray://...")
daft.context.set_runner_ray()
```

In this context, Ray actually starts a new cluster.

Unfortunately we currently use `self.ray_context` to determine whether we are running in "client mode".